### PR TITLE
Allow overriding of excluded children list from parent page

### PR DESCRIPTION
**Pull request by dhensby**
This change allows a parent page to supply a custom list of pages if the `getExcludedChildren` exists on the parent page.

also:
- More robust checking of being in CMS
- Less repeated code

### CODE REVIEW

**Comment by micschk** [comment]
Daniel, thanks!

File: code/ExcludeChildren.php [hunk 1]
```diff
@@ -22,30 +22,33 @@ public function getExcludedClasses(){
...
-		// Else; exclude children from CMS only
-		$action = Controller::curr()->getAction();
-		if (in_array($action, array("treeview", "listview", "getsubtree"))) {
+		$controller = Controller::curr();
+		$action = $controller->getAction();
+		if ($this->owner->config()->get("force_exclusion_beyond_cms")
+			|| ($controller instanceof LeftAndMain) && in_array($action, array("treeview", "listview", "getsubtree"))
+		) {
+			//if the page class has a getExcludedChildren function, use it to supply the list of children
+			if ($this->owner->hasMethod('getExcludedChildren')) {
```
Comment (giovdk21) [inline code/ExcludeChildren.php]: getExcludedChildren is misleading as we're actually returning the one that we're not excluding here

Comment (micschk) [inline code/ExcludeChildren.php]: @dhensby, how about we rename it to 'getExcludedChildClasses'?

**Comment by micschk** [comment]
@dhensby would you mind adding a note to the README explaining getExcludedChildren/getExcludedChildClasses?

**Comment by giovdk21** [comment]
Hello,
I worked with Dan (dhensby) on it and I can start explaining it here.

We wanted to give the user the control of what could be returned for a specific type of page but without completely overriding the getFilteredChildren method as we wanted the common logic to stay there.

So what is getExcludedChildren is doing is actually returning the filtered children from your Page type (so it's doing the opposite the name would suggest! :stuck_out_tongue_closed_eyes:)

One example is a specific PageHolder we had where we wanted to show on the tree only the children which are not marked as archived.

thanks.

File: code/ExcludeChildren.php (hunk 1 above)
Comment (dhensby) [inline code/ExcludeChildren.php]: well, this is meant to return a custom list of children to *show*... I'm not too hung up on names

**Comment by dhensby** [comment]
> @dhensby would you mind adding a note to the README explaining getExcludedChildren/getExcludedChildClasses?

see #10

**Pull request merged**